tutorials/text/deduplication/semantic/semantic_e2e.ipynb

```diff
@@ -0,0 +1,843 @@
+      "2025-09-16 11:59:52,919\tINFO worker.py:1789 -- Calling ray.init() again after it has already been called.\n",
+      "Fetching 30 files: 100%|██████████| 30/30 [00:00<00:00, 11242.77it/s]\n",
+      "Fetching 30 files: 100%|██████████| 30/30 [00:00<00:00, 29495.81it/s]\n",
+      "2025-09-16 12:03:06,910\tINFO worker.py:1630 -- Using address 127.0.1.1:6379 set in the environment variable RAY_ADDRESS\n",
+      "2025-09-16 12:03:06,912\tINFO worker.py:1771 -- Connecting to existing Ray cluster at address: 127.0.1.1:6379...\n",
+      "2025-09-16 12:03:06,919\tINFO worker.py:1942 -- Connected to Ray cluster. View the dashboard at \u001b[1m\u001b[32mhttp://127.0.0.1:8265 \u001b[39m\u001b[22m\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[36m(KMeansReadFitWriteStage pid=1854069)\u001b[0m 203520000\n",
+      "\u001b[36m(KMeansReadFitWriteStage pid=1854070)\u001b[0m 203323776\n",
+      "\u001b[36m(KMeansReadFitWriteStage pid=1854071)\u001b[0m 203520000\n",
+      "\u001b[36m(KMeansReadFitWriteStage pid=1854072)\u001b[0m 203520000\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2025-09-16 12:03:35,546\tINFO worker.py:1630 -- Using address 127.0.1.1:6379 set in the environment variable RAY_ADDRESS\n",
+      "2025-09-16 12:03:35,548\tINFO worker.py:1771 -- Connecting to existing Ray cluster at address: 127.0.1.1:6379...\n",
+      "2025-09-16 12:03:35,554\tINFO worker.py:1942 -- Connected to Ray cluster. View the dashboard at \u001b[1m\u001b[32mhttp://127.0.0.1:8265 \u001b[39m\u001b[22m\n",
+      "2025-09-16 12:03:35,581\tINFO worker.py:1630 -- Using address 127.0.1.1:6379 set in the environment variable RAY_ADDRESS\n",
+      "2025-09-16 12:03:35,583\tINFO worker.py:1771 -- Connecting to existing Ray cluster at address: 127.0.1.1:6379...\n",
+      "2025-09-16 12:03:35,583\tINFO worker.py:1789 -- Calling ray.init() again after it has already been called.\n",
+      "2025-09-16 12:04:02,497\tINFO worker.py:1630 -- Using address 127.0.1.1:6379 set in the environment variable RAY_ADDRESS\n",
+      "2025-09-16 12:04:02,498\tINFO worker.py:1771 -- Connecting to existing Ray cluster at address: 127.0.1.1:6379...\n",
+      "2025-09-16 12:04:02,505\tINFO worker.py:1942 -- Connected to Ray cluster. View the dashboard at \u001b[1m\u001b[32mhttp://127.0.0.1:8265 \u001b[39m\u001b[22m\n",
+      "2025-09-16 12:04:02,521\tINFO worker.py:1630 -- Using address 127.0.1.1:6379 set in the environment variable RAY_ADDRESS\n",
+      "2025-09-16 12:04:02,522\tINFO worker.py:1771 -- Connecting to existing Ray cluster at address: 127.0.1.1:6379...\n",
+      "2025-09-16 12:04:02,523\tINFO worker.py:1789 -- Calling ray.init() again after it has already been called.\n"
+     ]
+    }
+   ],
+   "source": [
+    "from nemo_curator.core.client import RayClient\n",
+    "\n",
+    "client = RayClient(num_cpus=64, num_gpus=4)\n",
```

Comment:
Are there any general rules of thumb for setting hardware requirements for this workflow, that the user should know?